File: test_data/typescript/test_main/aas_core_meta.v3/input/snippets/Verification/is_xs_date_time.ts
/**
 * Check that `value` is a valid `xs:dateTime`.
 *
 * @remarks
 * Year 1 BCE is the last leap BCE year.
 * See https://www.w3.org/TR/xmlschema-2/#dateTime.
 *
 * @param value - to be be checked
 * @returns `true` if `value` is a valid `xs:dateTime`
 */
export function isXsDateTime(value: string): boolean {
  // NOTE (mristin, 2022-11-23):
  // We can not use date functions from the standard library as it does not
  // handle years BCE (*e.g.*, `-0003-01-02`).

  if (!matchesXsDateTime(value)) {
    return false;
  }

  const date = value.split("T", 1)[0];
  return isXsDate(date);
}

File: test_data/python/test_main/aas_core_meta.v3/input/snippets/Verification/is_xs_date_time.py
def is_xs_date_time(value: str) -> bool:
    """
    Check that :paramref:`value` is a ``xs:dateTime`` with
    the time zone set to UTC.
    """
    if not matches_xs_date_time(value):
        return False

    date, _ = value.split("T")
    return is_xs_date(date)
